tests/xharness/AppRunner.cs

```diff
@@ -887,7 +923,7 @@ public async Task<int> RunAsync ()
 					FailureMessage = $"Launch failure";
 					if (Harness.InCI)
 						XmlResultParser.GenerateFailure (Logs, "launch", appName, Variation, $"AppLaunch on {device_name}", $"{FailureMessage} on {device_name}", main_log.FullPath, XmlResultParser.Jargon.NUnitV3);
-				} else if ((!File.Exists (listener_log.FullPath) || string.IsNullOrEmpty (crash_reason)) && Harness.InCI) {
+				} else if ((crashed && !File.Exists (listener_log.FullPath) || (!crashed && string.IsNullOrEmpty (crash_reason)) && Harness.InCI)) {
```

Comment:
`crashed` is not assigned (not used) elsewhere in the PR

Reply:
Not in the diff, but it is set here: https://github.com/xamarin/xamarin-macios/pull/7961/files#diff-03773861bcef485bf343300f31b60b0eL799

Noticed that we needed to check for that value, else we had some crashes wrongly reported. Can be in a diff PR.

Reply:
ok, if same-PR then make sure it's part of the PR/commit description. Would be easy to miss otherwise

Reply:
Added separate PR to unblock another PR: https://github.com/xamarin/xamarin-macios/pull/7979 will merge with master once it lands.